adapters/Unity/UnityAdapter/GADMAdapterUnityRewardedAd.m

```diff
@@ -14,224 +14,143 @@
-                                                                        delegate:self];
-}
-
-#pragma mark GADMAdapterUnityDataProvider Methods
-
-- (NSString *)getGameID {
-  return _gameID;
-}
-
-- (NSString *)getPlacementID {
-  return _placementID;
-}
-
-- (void)didFailToLoadWithError:(nonnull NSError *)error {
-  _adLoadCompletionHandler(nil, error);
+  [UnityAds show:viewController placementId:_placementID];
 }
 
 #pragma mark - Unity Delegate Methods
 
 - (void)unityAdsDidError:(UnityAdsError)error withMessage:(nonnull NSString *)message {
-  if (error == kUnityAdsErrorNotInitialized) {
-    if (_adLoadCompletionHandler) {
-      NSError *errorWithDescription =
-          GADMAdapterUnitySDKErrorWithUnityAdsErrorAndMessage(error, message);
-      _adLoadCompletionHandler(nil, errorWithDescription);
-      _adLoadCompletionHandler = nil;
-    }
-  } else {
-    if (_adEventDelegate) {
-      NSError *errorWithDescription =
-          GADMAdapterUnitySDKErrorWithUnityAdsErrorAndMessage(error, message);
-      [_adEventDelegate didFailToPresentWithError:errorWithDescription];
-    }
+  if (_adEventDelegate) {
+    NSError *errorWithDescription =
+    GADMAdapterUnitySDKErrorWithUnityAdsErrorAndMessage(error, message);
+    [_adEventDelegate didFailToPresentWithError:errorWithDescription];
   }
 }
 
 - (void)unityAdsDidFinish:(nonnull NSString *)placementID
           withFinishState:(UnityAdsFinishState)state {
-  if (![placementID isEqualToString:_placementID]) {
-    return;
-  }
-
+  
   if (state == kUnityAdsFinishStateCompleted) {
     [_adEventDelegate didEndVideo];
-
+    
     // Unity Ads doesn't provide a way to set the reward on their front-end. Default to a reward
     // amount of 1. Publishers using this adapter should override the reward on the AdMob
     // front-end.
     GADAdReward *reward = [[GADAdReward alloc] initWithRewardType:@""
                                                      rewardAmount:[NSDecimalNumber one]];
     [_adEventDelegate didRewardUserWithReward:reward];
   } else if (state == kUnityAdsFinishStateError) {
-    NSError *error = GADMAdapterUnityErrorWithCodeAndDescription(
-        GADMAdapterUnityErrorFinish,
-        @"UnityAds finished presenting with error state kUnityAdsFinishStateError.");
+    NSError *error = GADMAdapterUnityErrorWithCodeAndDescription(GADMAdapterUnityErrorFinish,
+                                                                 @"UnityAds finished presenting with error state kUnityAdsFinishStateError.");
     [_adEventDelegate didFailToPresentWithError:error];
   }
-
+  
   [_adEventDelegate willDismissFullScreenView];
   [_adEventDelegate didDismissFullScreenView];
 }
 
 - (void)unityAdsDidStart:(nonnull NSString *)placementID {
-  if ([placementID isEqualToString:_placementID]) {
-    [_adEventDelegate didStartVideo];
-  }
+  [_adEventDelegate didStartVideo];
 }
 
 - (void)unityAdsReady:(nonnull NSString *)placementID {
```

Comment:
Can you add a comment here to indicate that the adapter will rely on `unityAdsAdLoaded` to check if an ad has been loaded?